packages/web/src/components/gridView.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+  Copyright (c) Microsoft Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the 'License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import * as React from 'react';
+import { ListView } from './listView';
+import type { ListViewProps } from './listView';
+import './gridView.css';
+import { ResizeView } from '@web/shared/resizeView';
+
+export type Sorting<T> = { by: keyof T, negate: boolean };
+
+export type GridViewProps<T> = Omit<ListViewProps<T>, 'render'> & {
+  columns: (keyof T)[],
+  columnTitle: (column: keyof T) => string,
+  columnWidth: (column: keyof T) => number,
+  render: (item: T, column: keyof T, index: number) => React.ReactNode,
+  sorting?: Sorting<T>,
+  setSorting?: (sorting: Sorting<T> | undefined) => void,
+};
+
+export function GridView<T>(model: GridViewProps<T>) {
+  const initialOffsets: number[] = [];
+  for (let i = 0; i < model.columns.length - 1; ++i) {
+    const column = model.columns[i];
+    initialOffsets[i] = (initialOffsets[i - 1] || 0) + model.columnWidth(column);
```

Comment:
Do we want this computation to be done once instead of on every render, e.g. with `useMemo()`?

Reply:
I'm using it synchronously as a default value of the offsets state, so I guess I need to do it here.